### ISSUE TITLE
remove-statements action failing for SNS resource

### ISSUE THREAD

**Issue by kumarsiyer** [capitalone/cloud-custodian#1779]
While running the remove-statements action for sns resource 
Policy that am using

```
policies:
  - name: sns-cross-account
    resource: sns
    filters:
      - type: cross-account             
    actions:
      - type: remove-statements
        statement_ids: matched
```

> Traceback (most recent call last):
>   File "/Users/test/cloud-custodian/c7n/resources/sns.py", line 96, in process
>     results += filter(None, [self.process_resource(client, r)])
>   File "/Users/test/cloud-custodian/c7n/resources/sns.py", line 117, in process_resource
>     AttributeValue=json.dumps(p)
>   File "/usr/local/lib/python2.7/site-packages/botocore/client.py", line 312, in _api_call
>     return self._make_api_call(operation_name, kwargs)
>   File "/usr/local/lib/python2.7/site-packages/botocore/client.py", line 605, in _make_api_call
>     raise error_class(parsed_response, operation_name)
> InvalidParameterException: An error occurred (InvalidParameter) when calling the SetTopicAttributes operation: Invalid parameter: Policy Error: null

**Comment by kumarsiyer** [comment]
Update: Even though there is an error message the matching statements were removed.

**Comment by kumarsiyer** [comment]
@kapilt

**Comment by hav3wb** [comment]
@kumarsiyer @kapilt I ran into the same issue as well. It appears for SQS as well. If removing the last policy statement from a resource, it appears to break. If using matched for example, and it needs to remove all of the statements, then will return that error. I tried running on specific Sids and worked fine when removing 1 of 2 from a resource, but when I ran to remove 1 of 1 statements, it caused an error.
